workspaces/ui-v2/src/optic-components/common/EditableTextField.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { ChangeEvent, FC } from 'react';
+import { TextField, Typography } from '@material-ui/core';
+import makeStyles from '@material-ui/styles/makeStyles';
+import { OpticBlueReadable } from '../theme';
+
+type EditableTextFieldProps = {
+  isEditing: boolean;
+  setEditing: (isEditing: boolean) => void;
+  value: string;
+  setValue: (newValue: string) => void;
+  helperText?: string;
+  defaultText?: string;
+  variant: TextFieldVariant;
+};
+
+export enum TextFieldVariant {
+  SMALL,
+  REGULAR,
+}
+
+export const EditableTextField: FC<EditableTextFieldProps> = ({
```

Comment:
So this isn't very generic / reusable - but I figured the best way to start creating reusable components is to copy + paste + create variants until we know what a common pattern is (or when we want to implement some consistency around our UI / design). Basically I figured copy paste while just hard coding different variants until it becomes too crazy seems like a good middle ground here

Reply:
Yes! No abstraction > the wrong abstraction, so when not sure, copying code is the way to go :)